src/Symfony/Component/Validator/Constraints/BackedEnumValueValidator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * BackedEnumValueValidator validates that a backed enum case can be hydrated from a value.
+ *
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+class BackedEnumValueValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof BackedEnumValue) {
+            throw new UnexpectedTypeException($constraint, BackedEnumValue::class);
+        }
+
+        if (null === $value || '' === $value) {
```

Comment:
i think the empty string case should pass through https://3v4l.org/lu4o4

Reply:
but an empty value can be a valid case https://3v4l.org/lu4o4

Reply:
i think generally so, it's an edge case

Reply:
No, your "edge case" simply does not make a difference.

Reply:
allowing empty string, only when it's a valid enum case, is not possible currently

Reply:
see also Choice constraint https://github.com/symfony/symfony/blob/0f6b3efdfd9adbb69a34910c5e42c9a1c3db0b2b/src/Symfony/Component/Validator/Constraints/ChoiceValidator.php#L40-L42